store/tikv/mock-tikv/cluster.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mocktikv
+
+import (
+	"bytes"
+	"sync"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/pingcap/kvproto/pkg/metapb"
+)
+
+// Cluster simulates a TiKV cluster. It focuses on management and the change of
+// meta data. A Cluster mainly includes following 3 kinds of meta data:
+// 1) Region: A Region is a fragment of TiKV's data whose range is [start, end).
+//    The data of a Region is duplicated to multiple Peers and distributed in
+//    multiple Stores.
+// 2) Peer: A Peer is a replica of a Region's data. All peers of a Region forms
+//    a group, each group elects a Leader to provide services.
+// 3) Store: A Store is a storage/service node. Try to think it as a TiKV server
+//    process. It only responds to client's request if the request's Region's
+//    leader Peer is on the Store.
+type Cluster struct {
+	mu      sync.RWMutex
+	id      uint64
+	stores  map[uint64]*Store
+	regions map[uint64]*Region
+}
+
+// NewCluster creates an empty cluster. It needs to be bootstrapped before
+// service.
+func NewCluster() *Cluster {
+	return &Cluster{
+		stores:  make(map[uint64]*Store),
+		regions: make(map[uint64]*Region),
+	}
+}
+
+// AllocID creates an unique ID in cluster. The ID could be used as either
+// StoreID, RegionID, or PeerID.
```

Comment:
@shenli This style could help resolving ambiguity. https://en.wikipedia.org/wiki/Serial_comma#Resolving_ambiguity

But in our case I think both is OK :)